### PR TITLE
Added rename method

### PR DESCRIPTION
**Pull request by NetworkNed**
I'd be grateful if you would consider this contribution to the Perl Google Drive library. It adds a method allowing the renaming of files and folders on Google Drive.

### CODE REVIEW

File: eg/google-drive-rename [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/perl
+use strict;
+
+use feature ':5.28';
```
Comment (atoomic) [inline eg/google-drive-rename]: This is going to break on Perl versions < 5.28

I really like using modern features but we should consider preserving backward compatibility with Perl 5.10 and later

At first glance I'm not sure to notice any 5.28 feature in the code after
please remove the line and confirm this is working as expected

File: lib/Net/Google/Drive/Simple.pm [hunk 2]
```diff
@@ -262,6 +262,45 @@ sub file_upload {
     return $file_id;
 }
 
+###########################################
+sub rename {
+###########################################
+    my ( $self, $file_id, $new_name ) = @_;
+
```
Comment (atoomic) [inline lib/Net/Google/Drive/Simple.pm]: maybe want to check that `$file_id` and `$new_name` are provided by the caller

File: lib/Net/Google/Drive/Simple.pm [hunk 3]
```diff
@@ -262,6 +262,45 @@ sub file_upload {
...
+    my $url = URI->new( $self->{api_file_url} . "/$file_id" );
+
+    my $req = &HTTP::Request::Common::PUT(
```
Comment (atoomic) [inline lib/Net/Google/Drive/Simple.pm]: The `&` is not required there
`s{&HTTP::Request::Common::PUT}{HTTP::Request::Common::PUT}`
... I see you copy/paste it from another function

Comment (atoomic) [inline lib/Net/Google/Drive/Simple.pm]: would be a bonus to have one `http_put` helper and use it, similar to `http_delete`
you would pass one hashref to it which would be everything after  authorization_headers
`$self->http_put( $url, { Accept => "application/json", ... } )`

File: eg/google-drive-rename (hunk 1 above)
Comment (NetworkNed) [reply eg/google-drive-rename]: Excuse me. This was left over from `say` statements I used when debugging. It was necessary and I have removed it in my commit 542b534901b0e5aab8c9fddbc6801e04de195899.

File: lib/Net/Google/Drive/Simple.pm (hunk 2 above)
Comment (NetworkNed) [reply lib/Net/Google/Drive/Simple.pm]: Am I missing something, or do none of the other functions do that? E.g. `sub file_create` and `sub file_upload`?

I'm not saying you're wrong to desire it - I just copied the conventions of the existing code, or thought I did, and would rather add error checking to every sub as a separate patch if you want.

File: lib/Net/Google/Drive/Simple.pm (hunk 3 above)
Comment (NetworkNed) [reply lib/Net/Google/Drive/Simple.pm]: Happy to take a look at this. The function would be called by both `sub rename` and `sub file_upload`, so calls by the latter would involve passing the contents of the file to the helper sub.

I.e. we could easily have 100MB or 1GB of data in the hashref - are there any performance considerations that need to be considered? I have no idea - that's why I'm asking.

File: lib/Net/Google/Drive/Simple.pm (hunk 2 above)
Comment (atoomic) [reply lib/Net/Google/Drive/Simple.pm]: I understand that you mimic the other functions. Was just making general comments to the provided code. The same comments are valid for the other functions too.
Overtime we are all improving the way we code, we should not hide behind bad habits.

Comment (atoomic) [reply lib/Net/Google/Drive/Simple.pm]: a separate patch is fine if you want

File: lib/Net/Google/Drive/Simple.pm (hunk 3 above)
Comment (atoomic) [reply lib/Net/Google/Drive/Simple.pm]: as you said it s a `dataref` which means it's a `ref`/`reference` to the hash... there is only one single hash. The reference is just a pointer to the data structure, there is no harms in passing pointers. Usually we want to limit the number of arguments as in most language for performance purpose, and avoiding spamming the stack.

Comment (NetworkNed) [reply lib/Net/Google/Drive/Simple.pm]: Please see my commit 5728e16aa8e9413bed5146d37ec5d49a7697db60

**Pull request merged**